Cod3rsGrowth.Web/wwwroot/Webapp/app/vendas/AdicionarVenda.controller.js

```diff
@@ -0,0 +1,343 @@
+            this._removerMessageStrip();
+            this._obterVendaPorId(oEvent);
+            var oRouter = this.getRouter();
+            _rota = oRouter.getRoute(oEvent.getParameter(parametroNome))._oConfig.name;
+
+        },
+
+        _carregarEventosCriar(oEvent) {
+            this._mudarTituloDaViewCriar();
+            this._limparInputs();
+            this._carregarCarros();
+            this._removerMessageStrip();
+            var oRouter = this.getRouter();
+            _rota = oRouter.getRoute(oEvent.getParameter(parametroNome))._oConfig.name;
+        },
+
+        _limparInputs() {
+            const inputNome = this.oView.byId(idDoInputNome);
+            const nome = inputNome.setValue(null);
+            inputNome.setValueState(sap.ui.core.ValueState.None);
+            inputNome.setValueStateText('');
+            const inputCpf = this.oView.byId(idDoInputCpf);
+            const cpf = inputCpf.setValue(null);
+            inputCpf.setValueState(sap.ui.core.ValueState.None);
+            inputCpf.setValueStateText('');
+            const inputEmail = this.oView.byId(idDoInputEmail);
+            const email = inputEmail.setValue(null);
+            inputEmail.setValueState(sap.ui.core.ValueState.None);
+            inputEmail.setValueStateText('');
+            const inputTelefone = this.oView.byId(idDoInputTelefone);
+            const telefone = inputTelefone.setValue(null);
+            inputTelefone.setValueState(sap.ui.core.ValueState.None);
+            inputTelefone.setValueStateText('');
+            const pago = this.oView.byId(idDoInputPago).setState(false)
+        },
+
+        _removerMessageStrip(){
+            this.getView().byId(idDoMessageStripSucessoCriacao).setVisible(false);
+            this.getView().byId(idDoMessageStripErroCriarVenda).setVisible(false);
+            this.getView().byId(idDoMessageStripSucessoEdicao).setVisible(false);
+            this.getView().byId(idDoMessageStripErroEditarVenda).setVisible(false);
+            this.oView.byId(idObjectStatusCarroNaoSelecionado).setVisible(false);
+        },
+
+        _carregarCarros() {
+            let sucesso = true;
+            fetch(urlCarrosDisponiveis)
+                .then((res) => {
+                    if (!res.ok) {
+                        sucesso = false;
+                    }
+                    return res.json()
+                })
+                .then((carro) => {
+                    const jsonModel = new JSONModel(carro)
+
+                    sucesso ? this.getView().setModel(jsonModel, modeloCarro)
+                        : this._erroNaRequisicaoDaApi(carro);
+                })
+                .catch((err) => MessageBox.error(err));
+        },
+
+        _mudarTituloDaViewCriar() {
+            var Titulo = this.getView().byId(idDoTituloTela).setText(textoParaAdicionarTituloCriar);
+            return Titulo;
+        },
+
+        aoColetarNome() {
+            const inputNome = this.oView.byId(idDoInputNome);
```

Comment:
muda para obterNome e remove essa variavel e faz direto 
const nome = this.oView.byId(idDoInputNome).getValue();